Translation/PlatformUIExtractor.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformUIBundle\Translation;
+
+use Symfony\Component\Translation\Extractor\AbstractFileExtractor;
+
+abstract class PlatformUIExtractor extends AbstractFileExtractor
+{
+    private $pathSuffix = 'Resources/views';
+
+    /**
+     * Updates the resource by replacing the path to templates (Resource/views)
+     * with the path handled by the extractor.
+     *
+     * @param string $resource
+     * @param string $path The path 'Resources/views' should be replaced with
+     */
+    protected function updateResource(&$resource, $path)
+    {
+        if (is_string($resource) && strpos($resource, $this->pathSuffix) !== null) {
```

Comment:
I'd keep the `is_array`/`is_string` test in `updateResource`, but move the `strpos` one to `replacePath`. Then we would have the best of both worlds, I think. I'll write it down and see how it looks like (moving this to #721)